companion/src/modeledit/flightmodes.cpp

```diff
@@ -160,21 +160,24 @@ FlightModePanel::FlightModePanel(QWidget * parent, ModelData & model, int phaseI
 
     // Column headers
     int headerCol = 1;
-    QLabel *nameLabel = new QLabel(ui->gvGB);
-    nameLabel->setText(tr("Name"));
-    gvLayout->addWidget(nameLabel, 0, headerCol++, 1, 1);
+
+    if (firmware->getCapability(GvarsName)) {
+      QLabel *nameLabel = new QLabel(ui->gvGB);
+      nameLabel->setText(tr("Name"));
+      gvLayout->addWidget(nameLabel, 0, headerCol++, 1, 1);
+    }
 
     if (phaseIdx > 0) {
       QLabel *sourceLabel = new QLabel(ui->gvGB);
       sourceLabel->setText(tr("Value source"));
       gvLayout->addWidget(sourceLabel, 0, headerCol++, 1, 1);
     }
 
-    if (IS_HORUS_OR_TARANIS(board) && phaseIdx == 0) {
-      QLabel *valueLabel = new QLabel(ui->gvGB);
-      valueLabel->setText(tr("Value"));
-      gvLayout->addWidget(valueLabel, 0, headerCol++, 1, 1);
+    QLabel *valueLabel = new QLabel(ui->gvGB);
+    valueLabel->setText(tr("Value"));
+    gvLayout->addWidget(valueLabel, 0, headerCol++, 1, 1);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Extract the assignment from this expression. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=cpp%3AAssignmentInSubExpression)